src/main/java/org/broadinstitute/hellbender/utils/GATKProtectedMathUtils.java

```diff
@@ -136,4 +141,22 @@ public static double stdDev(final double ... values) {
         Utils.nonNull(values);
         return Math.sqrt(new Variance().evaluate(values));
     }
+
+    // given a list of options and a function for calculating their probabilities (these must sum to 1 over the whole list)
+    // randomly choose one option from the implied categorical distribution
+    public static <E> E randomSelect(final List<E> choices, final Function<E, Double> probabilityFunction) {
```

Comment:
Since this is pretty standard, it seems like you should make use of EnumeratedDistribution in Apache Commons (or something along those lines) here.  You could then pass a RandomGenerator (which you could reset before each test/method that calls this, avoiding those problems on Travis that we ran into before), rather than using the private static Random (which you cannot reset in this code as written).